tools/start_server.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import os.path
+import re
+import select
+import shutil
+import subprocess
+import sys
+import tarfile
+from pathlib import Path
+from subprocess import check_call
+from tempfile import TemporaryDirectory
+
+from hypy_utils import ensure_dir
+from hypy_utils.downloader import download_file
+
+from .download_jdk import download_oracle, ensure_java
+
+MAX_RAM = '16384M'
+MIN_RAM = MAX_RAM
+TIME = 5
+
+version_re = re.compile(r'[0-9].[0-9]+.[0-9]+')
+
+
+def start(java: Path, mc_path: Path):
+    # Find server jar
+    jar = ''
+    for f in [str(f) for f in os.listdir(mc_path) if f.endswith('.jar')]:
+        ver = version_re.findall(f)
+        if any(k in f.lower() for k in ['craftbukkit', 'spigot', 'paper', 'purpur']) or len(ver) != 0:
+            jar = f
+            break
+
+    # Auto Restart
+    while True:
+        cmd = f"{java.absolute()} -Xmx{MAX_RAM} -Xms{MIN_RAM} --add-modules=jdk.incubator.vector -jar {jar} nogui"
+        print(f'Executing {cmd}...')
+        subprocess.Popen(cmd, shell=True, cwd=mc_path).wait()
+
+        print('Server stopped, restarting in 5s\nPress any key to stop the server.')
+        i, o, e = select.select([sys.stdin], [], [], 5)
+        if i:
```

Comment:
Can you name the 3 vowel variables? I'm guessing this line is grabbing keyboard input w/ a timeout of 5s, but with the output being unnamed I have to go to the select module to grab a better understanding of what this is doing.

Reply:
Ok just changed the variable names